### ISSUE TITLE
Connection to MQTT server failed : [Errno 99] Address not available

### ISSUE THREAD

**Issue by smathev** [nicolargo/glances#2063]
**Describe the bug**
Glances is no longer connecting to the mqtt-broker. Nothing changed in my setup, except I updated to the latest-full version from 15 days ago.

**To Reproduce**
Use an mqtt-broker, and set glances to connect to that. You should see what I'm seeing.

**Expected behavior**
I expected glances to connect to the mqtt-broker like it did in the version before this one.

DockerImage: latest-full, alpine-latest-full and dev all reveal the same issue
Distro...: Debian GNU/Linux 11 (bullseye)
Kernel...: Linux 5.10.0-13-amd64

**Additional context**
Log output from latest-full and dev:
`Connection to MQTT server failed : [Errno 99] Cannot assign requested address `
Repeating after every restart.

Log output from alpine-latest-full:
```
Error while initializing the docker plugin (No module named 'six')
Connection to MQTT server failed : [Errno 99] Address not available
Exception in thread Thread-6 (update):
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1009, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.10/threading.py", line 946, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.10/site-packages/glances/exports/glances_export.py", line 169, in update
    all_stats = stats.getAllExportsAsDict(plugin_list=self.plugins_to_export())
  File "/usr/lib/python3.10/site-packages/glances/stats.py", line 274, in getAllExportsAsDict
    return {p: self._plugins[p].get_export() for p in plugin_list}
  File "/usr/lib/python3.10/site-packages/glances/stats.py", line 274, in <dictcomp>
    return {p: self._plugins[p].get_export() for p in plugin_list}
AttributeError: 'dict' object has no attribute 'get_export'
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/usr/lib/python3.10/site-packages/glances/__main__.py", line 29, in <module>
    glances.main()
  File "/usr/lib/python3.10/site-packages/glances/__init__.py", line 189, in main
    start(config=core.get_config(), args=core.get_args())
  File "/usr/lib/python3.10/site-packages/glances/__init__.py", line 141, in start
    mode.serve_forever()
  File "/usr/lib/python3.10/site-packages/glances/standalone.py", line 173, in serve_forever
    while self.__serve_once():
  File "/usr/lib/python3.10/site-packages/glances/standalone.py", line 139, in __serve_once
    self.stats.update()
  File "/usr/lib/python3.10/site-packages/glances/stats.py", line 216, in update
    if self._plugins[p].is_disabled():
AttributeError: 'dict' object has no attribute 'is_disabled'
```

**Comment by nicolargo** [comment]
Hi @smathev 

The first error is "**Connection to MQTT server failed : [Errno 99] Address not available**"

Can you please provide more information:

- MQTT server
- Glances configuration file

Nicolas

We need more information

**Comment by nicolargo** [comment]
Should be related to: https://github.com/nicolargo/glances/issues/2061

**Comment by smathev** [comment]
**MQTT server:**
Using a docker-cotainer. `Eclipse-Mosquitto:latest` HomeAssistant and Zigbee2MQTT connects without issue. Have tried restarting both Eclipse and Glances multiple times.

**Glances docker_compose file:**
```
monitoring:
  image: nicolargo/glances:latest-full
  container_name: glances
  restart: always
  ports:
    - "61208-61209:61208-61209"
  volumes:
	- /var/run/docker.sock:/var/run/docker.sock
	- /home/core/docker_volumes/glances/:/glances/conf/
  environment:
	- "GLANCES_OPT=--disable-process --quiet --export mqtt"
```


**Glances Configuration file:**
```
[mqtt]
# Configuration for the --export mqtt option
host=(my static IP provided from my router) 192.168.X.X
port=1883
tls=false
user="mqtt-username"
password="mqtt-password"
topic=glances
topic_structure=per-metric
```
I've tried resetting my username and password in the MQTT server also, to see if that provided any solution.

I hope this is sufficient, in terms of additional troubleshooting. Let me know if you need anything else, or if I need to switch to `:dev `tag in glances-docker.

**Comment by smathev** [comment]
Just as an update to this - I have tried using the latest dev-build and it failed starting, kept restarting - writing this to "docker logs glances" over and over:

```
Connection to MQTT server localhost:8883 failed with error: [Errno 99] Cannot assign requested address
MQTT client initialization failed
Connection to MQTT server localhost:8883 failed with error: [Errno 99] Cannot assign requested address
MQTT client initialization failed
```

which is weird, cause my configuration file uses port 1883 and host is not localhost. So, I figured it might not be using my configuration-file.

I then tried adding:
`-C /glances/conf/glances.conf`
as additional enviroments options in my docker-compose file. 

That seems to have done the trick - my glances-docker is now functional again. I'm not sure what code has been updated, but it solved my issue - once I directly told the docker where my config file was. For reference, it was always placed in /glances/conf/glances.conf on the docker and I always had that as a mounted volume on my local disk.

Anyway, this issue is closed.
